### PR TITLE
ControlAllocation: implement custom saturation logic for tilt-for-yaw actuation

### PR DESCRIPTION
**Pull request by sfuhrer**
Addresses https://github.com/PX4/PX4-Autopilot/issues/20395

## Describe problem solved by this pull request
There is very often some left over (unallocated) yaw torque, even though the tilts are not saturating. Reason is that when the tilts are used for yaw control, they are meant to move symmetrically to the front and back, but back is limited and thus there is some leftover torque not allocated. 

## Describe your solution
- in `ControlAllocator::publish_control_allocator_status()` always fill allocated/unallocated torque from matrix, then allow them to be overwritten by custom `getAllocatedAndUnallocatedControl() `implementation (specific for effectiveness type)
- add `actuatorEffectivenessTiltrotorVTOL::getAllocatedAndUnallocatedControl()` that only affects the yaw unallocated torque of matrix 0 (titls). Set `unallocated_torque[2] `corresponding to if the current yaw torque setpoint is within or outside of range (-1, 1).
- `control_allocator_status.torque_setpoint_achieved` is always set to `if(norm(unallocated_torque) < 1e-6)`, not in custom saturation implementation. 
- remove logging of `allocated_torque` and `allocated_thrust`, as redundant information that can also be get through `setpoint`-`unallocated` 

## Describe possible alternatives
Custom tilt allocation (instead of only custom saturation logic). 

## Test data / coverage
SITL tiltrotor tested. 

## Additional context
In https://github.com/PX4/PX4-Autopilot/commit/cf8262465d7a23c125f6d60ef8dd7d0c545a41ab I simply renamed a couple of variables to make it easier to distinguish collective_tilt and separate tilt.

### CODE REVIEW

File: src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessTiltrotorVTOL.hpp [hunk 1]
```diff
@@ -93,8 +96,15 @@ class ActuatorEffectivenessTiltrotorVTOL : public ModuleParams, public ActuatorE
...
 
 	uORB::Subscription _actuator_controls_1_sub{ORB_ID(actuator_controls_1)};
 	uORB::Subscription _actuator_controls_0_sub{ORB_ID(actuator_controls_0)};
+
+	struct YawTiltSaturationFlags {
+		bool tilt_yaw_pos;
+		bool tilt_yaw_neg;
+	};
+
+	YawTiltSaturationFlags _yaw_tilt_saturation_flags;
```
Comment (bkueng) [inline src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessTiltrotorVTOL.hpp:109]:
```suggestion
	YawTiltSaturationFlags _yaw_tilt_saturation_flags{};
```

File: src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectiveness.hpp [hunk 2]
```diff
@@ -199,10 +199,10 @@ class ActuatorEffectiveness
 	virtual uint32_t getStoppedMotors() const { return 0; }
 
 	/**
-	 * Fill in the allocated and unallocated torque and thrust.
-	 * Should return false if not filled in and the effectivenes matrix should be used instead
+	 * Fill in the unallocated torque and thrust, customized by effectiveness type.
+	 * Can be implemented for every type separately. If not implemented then the effectivenes matrix is used instead.
 	 */
-	virtual bool getAllocatedAndUnallocatedControl(control_allocator_status_s &status) const { return false; }
+	virtual void getUnallocatedControl(control_allocator_status_s &status, int matrix_index) {}
```
Comment (bkueng) [inline src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectiveness.hpp:205]: Can you move output arguments last?
```suggestion
	virtual void getUnallocatedControl(int matrix_index, control_allocator_status_s &status) {}
```

**Comment by sfuhrer** [comment]
Flight tested, no issues noted: https://review.px4.io/plot_app?log=36dbecb8-3669-4895-863f-6f264bc3d8d9

**Comment by bkueng** [comment]
Can you add the `Apply suggestions from code review` commit to the relevant commits before?

**Comment by sfuhrer** [comment]
> Can you add the `Apply suggestions from code review` commit to the relevant commits before?

done!

**Review by bkueng — APPROVED** [review]
Thanks

**Pull request merged**